text/1015-full-text-search.rst

```diff
@@ -73,59 +73,57 @@ queries rather than introducing a new scalar type. This string is expected to
 contain FTS search queries in a format similar to how they might be typed into
 a search field.
 
-The searching functionality can be split along two axis:
-1) what FTS operates on (``str`` vs ``Object``)
-2) what FTS returns (``bool`` vs some ranked results)
+The searching functionality differs based on whether the result is a ``bool``
+indicating a match, or the result is providing some ranking information and
+potentially match highlighting.
 
 The task of testing whether a document matches an FTS query would be performed
-by the following built-in functions::
-
-  function fts::test(doc: str, query: str) -> bool
-  function fts::test(doc: Object, query: str) -> bool
-
-The input can be either a ``str`` or an entire ``Object``. The object type
-must have some ``str`` properties, however. Trying to perform FTS on a type
-without ``str`` properties is an error. This function is generally intended to
-be used in ``filter`` clauses in cases where there's no need for hinting what
-matched, but the results are simple enough to be self-explanatory.
-
-The ranked results need to be rich objects with several relevant properties,
-akin to free objects used for returning ``group`` results. The difference is
-that we cannot use a ``FreeObject`` as a return type of a function, we need a
-specialized type for that: ``FTSResult``. This type is defined as follows::
-
-  type fts::FTSResult {
-    property rank -> float64 {
-      default := 0;
-    }
-    property highlights -> array<str> {
-      default := [];
-    }
-  }
-
-If ranking was requested, then the value will be written in ``rank``.
+by the following built-in function::
+
+  function std::fts::test(
+      query: str,
+      variadic doc: optional str,
+      named only language: str,
+  ) -> bool
+
+The ``doc`` input can be one or more ``str`` values, which allows passing
+multiple properties of an object to be searched. This function is generally
+intended to be used in ``filter`` clauses in cases where there's no need for
+hinting what matched, but the results are simple enough to be
+self-explanatory.
+
+The ``language`` parameter specifies which language should be used by the FTS
+engune when processing the ``query`` and the ``doc``. A possible additional
+option is to have a ``global std::fts::language -> str`` which then can be
+used as the default for this parameter. Alternatively, the default language
+may be set via config values. If either of these options is implemented, the
+``language`` parameter may be omitted.
+
+The ranked results need to have richer information and thus named tuples are
+used::
+
+  function std::fts::match(
+      query: str,
+      variadic doc: optional str,
```

Comment:
(`variadic` + `optional` is definitely workable for a built-in SQL defined function like `fts::match` but seeing this made me realize that we allow it in edgeql and it does not behave sensibly (https://github.com/edgedb/edgedb/issues/4982))